internal/api/project.go

```diff
@@ -8,20 +8,41 @@ import (
 
 type Project struct {
 	ID            string
+	Name          string
 	Slug          string
 	DefaultParams map[string]interface{}
 }
 
 func GetProject(client graphql.Client, orgID string, idOrSlug string) (*Project, error) {
 	response, err := getProjectById(context.Background(), client, orgID, idOrSlug)
+	p := response.Project.toProject()
+	return &p, err
+}
 
-	return response.Project.toProject(), err
+func (p *getProjectByIdProject) toProject() Project {
+	return Project{
```

Comment:
@chrisghill this method 'casts' the graphql representation of a project (of the current query) to an API Project ... 

graphql can have many representations of a project (different fields depending on query) ... this allows us to have one definition of a project that gets returned to API callers...

Is there a way to write a struct function on an interface?

Like can we say

```go
type Projectish interface{
  Id string
  Slug String
  // ...
}
func (i *Projectish) toProject() Project {
  return Project{
    ID: i.Id, //...
  }
}
```